src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -479,14 +521,19 @@ private void BeginCondenseTransition()
             if (this.IsExpanded == false)
                 throw new InvalidOperationException();
 
-            RefreshCondensedDisplay();
+            //To prevent another transition from being started
+            currentState = State.InTransition;
 
-            this.smallContentGrid.Visibility = System.Windows.Visibility.Visible;
-            SetCurrentStateAndNotify(State.InTransition);
+            RefreshCondensedDisplay(() =>
+                {
+                    this.smallContentGrid.Visibility = System.Windows.Visibility.Visible;
+                    SetCurrentStateAndNotify(State.InTransition);
```

Comment:
Thanks for the point! I feel it is more safe to set the currentState directly above and remove the line here. The notify part will cause OnPreviewControlStateChanged to be executed which will start a new transition.